File: api/DTOs/Finance/StatementofAccountDto.cs
namespace api.DTOs.Finance
{
    public class StatementOfAccountDto
    {
        public int AccountId { get; set; }
        public string AccountName { get; set; }
		public DateOnly FromDate {get; set;}
		public DateOnly UptoDate {get; set;}
		public long OpBalance {get; set;}
		public long ClBalance {get; set;}	//balance for the period
        public ICollection<StatementOfAccountItemDto> StatementOfAccountItems { get; set; }
    }
}

File: client/src/app/_dtos/finance/statementOfAccountDto.ts
export interface IStatementofAccountDto
{
	accountId: number;
	accountName: string;
	fromDate: Date;
	uptoDate: Date;
	opBalance: number;
	clBalance: number;
	
	statementOfAccountItems: IStatementofAccountItemDto[];

}

export interface IStatementofAccountItemDto
{
	voucherNo: number;
	transDate: Date;
	coaId: number;
	accountName: string;
	dr: number;
	cr: number;
	narration: string;

}